packages/core/src/test/shared/ui/sam/stackPrompter.test.ts

```diff
@@ -0,0 +1,131 @@
+import * as SamUtilsModule from '../../../../shared/sam/utils'
+import * as ButtonsModule from '../../../../shared/ui/buttons'
+import { DefaultCloudFormationClient } from '../../../../shared/clients/cloudFormationClient'
+import { samSyncUrl } from '../../../../shared/constants'
+import { createStackPrompter } from '../../../../shared/ui/sam/stackPrompter'
+import { intoCollection } from '../../../../shared/utilities/collectionUtils'
+
+describe('createStackPrompter', () => {
+    let sandbox: sinon.SinonSandbox
+    const cfnClient = new DefaultCloudFormationClient('us-east-1')
+    const mementoRootKey = 'samcli.sync.params'
+
+    beforeEach(() => {
+        sandbox = sinon.createSandbox()
+    })
+
+    afterEach(() => {
+        sandbox.restore()
+    })
+
+    it('should create a prompter with existing stacks', async () => {
+        // Arrange
+        const stackSummaries: CloudFormation.StackSummary[][] = [
+            [
+                {
+                    StackName: 'stack1',
+                    StackStatus: 'CREATE_COMPLETE',
+                    CreationTime: new Date(),
+                } as CloudFormation.StackSummary,
+                {
+                    StackName: 'stack2',
+                    StackStatus: 'CREATE_COMPLETE',
+                    CreationTime: new Date(),
+                } as CloudFormation.StackSummary,
+                {
+                    StackName: 'stack3',
+                    StackStatus: 'CREATE_COMPLETE',
+                    CreationTime: new Date(),
+                } as CloudFormation.StackSummary,
+            ],
+        ]
+        const expectedItems = [
+            {
+                label: 'stack1',
+                data: 'stack1',
+                description: undefined,
+                invalidSelection: false,
+                recentlyUsed: false,
+            },
+            {
+                label: 'stack2',
+                data: 'stack2',
+                description: undefined,
+                invalidSelection: false,
+                recentlyUsed: false,
+            },
+            {
+                label: 'stack3',
+                data: 'stack3',
+                description: undefined,
+                invalidSelection: false,
+                recentlyUsed: false,
+            },
+        ]
+        const listAllStacksStub = sandbox.stub(cfnClient, 'listAllStacks').returns(intoCollection(stackSummaries))
+        sandbox.stub(SamUtilsModule, 'getRecentResponse').returns(undefined)
+        const createCommonButtonsStub = sandbox.stub(ButtonsModule, 'createCommonButtons')
+        sandbox
+            .stub(AwsConsoleModule, 'getAwsConsoleUrl')
+            .returns(vscode.Uri.parse(`https://us-east-1.console.aws.amazon.com/cloudformation/home?region=us-east-1`))
+
+        // Act
+        const prompter = createStackPrompter(cfnClient, mementoRootKey, samSyncUrl)
+        await new Promise((f) => setTimeout(f, 50))
```

Comment:
Use `waitUntil`. Applicable throughout

Reply:
Understood. After checking with the test author, a more appropriate option is to use `await sleep()`.

Reply:
Will update to using `await sleep()`.